### DISCUSSION TITLE
Integrating F in an individual/population simulation

### DISCUSSION THREAD

**Discussion by yahoff** [metrumresearchgroup/mrgsolve#1121] [Q&A] [answered]
Hello everyone,

I am currently working with mrgsolve to simulate PK models. One model has several covariates for bioavailability (F) and I would like to include it in my simulation. For an individual simulation it is quite straightforward. However, I would like to do it for a population as well.  I create a virtual population in r with specific ranges in a uniform distribution (e.g. bodyweight 50-150 kg, hematocrit 30-40%...). I use the PK model textfile and $PARAM to introduce the PK parameter. Can someone help me out how to introduce (and code) the bioavailability for the population to cover the whole range in the population?

Best,
Yannick

**Comment by FelicienLL** [comment, marked as answer]
Hi, 
In the following example, I simulate from a pre-generated population thanks to  `idata_set()`. Does it answer your question? 
Best regards.
``` r
library(mrgsolve)
mod <- mcode("mod", "
$PARAM BW = 70, SEX = 0
$MAIN
double CL = 1 * pow(BW/70, .75) * pow(.8, SEX) ;
double V = 10 ;
$PKMODEL ncmt = 1
$CMT CENT")
#> Building mod ...
#> done.

individuals <- expand.idata(BW = seq(30, 100, 10), SEX = c(0, 1))
mod %>% 
  idata_set(individuals) %>% 
  ev(amt = 100) %>% 
  mrgsim() %>% 
  plot()
```

![](https://i.imgur.com/M1KsFIh.png)

<sup>Created on 2023-10-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by yahoff** [reply]
Hello Felicien, thank you for your answer. I approached it in a similar way by using the expand.grid command to generate the population, so good to see a more straightforward approach by using expand.idata, I will try that as well!. I have two follow-up questions:
1. How do you integrate the bioavailability in your model in the case you showed?
2. Where can you include the amount of samples for monte carlo simulation and where do you sample from your IIV?

Best,
Yannick

**Comment by FelicienLL** [comment]
Yannick, 
1. Bioavailability is controled by the `F_CMT` term, see https://mrgsolve.org/user-guide/specification.html#sec-spec-f.
2. If you use `idata_set()` it is up to you to pass a table with the adequate number of patients inside, whatever R function you use to generate your population (`sample()`, `rnorm()`, `runif()`...). Alternatively, see `mrgsim(n = )` (but won't work with `idata_set()` as far as I remember). As in NONMEM, IIV can be set in a `$OMEGA` (https://mrgsolve.org/user-guide/specification.html#sec-block-omega) block and called with `ETA` in the code (https://mrgsolve.org/user-guide/specification.html#etan).

**Reply by yahoff** [reply]
Thank you for the clarification. I will give it a try with your suggestions!
